### PR TITLE
Issue 18

### PR DESCRIPTION
**Pull request by andrewgdotcom**

### CODE REVIEW

**Comment by andrewgdotcom** [comment]
Closes #18

File: constructor.sh [hunk 1]
```diff
@@ -148,13 +164,15 @@ TMP_DIR=$(mktemp -d)
...
-	adduser --disabled-login --system --home $BASE $USER
+if ! grep -q "^${ES_USER}:" /etc/passwd; then
+    proxy_log "adding user $ES_USER"
+	adduser --disabled-login --system --home $BASE $ES_USER
 fi
 
 # sometimes (I'm looking at you, Hetzner) we can find ourselves with a
 # bad IPv6 configuration. If so, disable it here.
 if [[ $DISABLE_IPV6 ]]; then
+    proxy_log "disabling ipv6"
```
Comment (scampi) [inline constructor.sh]: The indentation in this file could be fixed. I think a modeline at the bottom could help

http://vim.wikia.com/wiki/Modeline_magic

File: post-assembly.sh [hunk 2]
```diff
@@ -11,13 +11,46 @@
...
+            if [[ ${status[$slave]} != alive ]]; then
+                echo "$slave is up on port $ES_PORT"
+                status[$slave]=alive
+            fi
+        else
+            if [[ ${status[$slave]} == alive ]]; then
+                echo "$slave has stopped responding on $ES_PORT"
+                status[$slave]=unknown
+            fi
+            # if ! nc -w 5 $slave 22 </dev/null >/dev/null; then
```
Comment (scampi) [inline post-assembly.sh]: can this be removed ?

File: spawner.sh [hunk 3]
```diff
@@ -153,22 +186,34 @@ done
...
+for slave in ${SLAVE_IPS[@]}; do
 	while ! nc -w 5 $slave 22 </dev/null >/dev/null; do
 		sleep 5
 	done
-	echo "$slave running"
+	echo "ssh running on $slave"
 done
 # Repopulate known_hosts
-ssh-keyscan $SLAVES >> $HOME/.ssh/known_hosts
+#ssh-keyscan ${SLAVE_IPS[@]} >> $HOME/.ssh/known_hosts
```
Comment (scampi) [inline spawner.sh:213]: should this removed ?

**Review by scampi — CHANGES_REQUESTED** [review]

File: constructor.sh (hunk 1 above)
Comment (andrewgdotcom) [reply constructor.sh]: I'm using Atom editor at the moment. Modelines only work for vim I think. Atom *should* fix indentation automatically, but it only does so consistently for new text.

Is there not a continuous-integration trick we can use in github? That would make it client-agnostic...

File: spawner.sh (hunk 3 above)
Comment (andrewgdotcom) [reply spawner.sh:213]: I'd really like to put this functionality back in iff I can find a way to do it without taking out the network. :-( Is it OK if I leave it there for now?

File: constructor.sh (hunk 1 above)
Comment (andrewgdotcom) [reply constructor.sh]: Fixed in this instance, but we should look at something more automated

File: post-assembly.sh (hunk 2 above)
Comment (andrewgdotcom) [reply post-assembly.sh]: Removed. It was a nice feature to give me more feedback on the behaviour of the slaves during construction, but it ended up causing the problem I was trying to solve.. ;-)

File: spawner.sh (hunk 3 above)
Comment (scampi) [reply spawner.sh:213]: sure!

**Review by scampi — APPROVED** [review]

**Pull request merged**